src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,32 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter format = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate parsedMinDate = LocalDate.parse(dateFrom, format);
+        LocalDate parsedMaxDate = LocalDate.parse(dateTo, format);
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        for (String name : names) {
+            int totalIncome = 0;
+            for (String value : data) {
+                String[] valueArr = value.split(" ");
+                DateTimeFormatter formatDateOfValue = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+                LocalDate parsedDateOfValue = LocalDate.parse(valueArr[0], formatDateOfValue);
+                if (parsedDateOfValue.compareTo(parsedMinDate) >= 0
```

Comment:
[Checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#remember-about-informative-names-of-the-variables)